src/main/java/com/cburch/logisim/std/ttl/Ttl74192.java

```diff
@@ -0,0 +1,243 @@
+   */
+  public static final String _ID = "74192";
+
+  public static final int PORT_INDEX_B = 0;
+  public static final int PORT_INDEX_QB = 1;
+  public static final int PORT_INDEX_QA = 2;
+  public static final int PORT_INDEX_DOWN = 3;
+  public static final int PORT_INDEX_UP = 4;
+  public static final int PORT_INDEX_QC = 5;
+  public static final int PORT_INDEX_QD = 6;
+  public static final int PORT_INDEX_D = 7;
+  public static final int PORT_INDEX_C = 8;
+  public static final int PORT_INDEX_LOAD = 9;
+  public static final int PORT_INDEX_CARRY = 10;
+  public static final int PORT_INDEX_BORROW = 11;
+  public static final int PORT_INDEX_CLEAR = 12;
+  public static final int PORT_INDEX_A = 13;
+  private static final String[] PORT_NAMES = {
+      "Data Input A",
+      "Data Output B",
+      "Data Output A",
+      "Count Down",
+      "Count Up",
+      "Data Output C",
+      "Data Output D",
+      "Data Input D",
+      "Data Input C",
+      "Load",
+      "Carry",
+      "Borrow",
+      "Clear",
+      "Data Input A"
+  };
+  private static final byte[] OUTPUT_PORTS = {2, 3, 6, 7, 12, 13};
+
+  private final int maxVal;
+
+  private Value cntD_prev = Value.NIL;
+  private Value cntU_prev = Value.NIL;
+
+  public Ttl74192() {
+    super(_ID, (byte) 16, OUTPUT_PORTS, PORT_NAMES, null);
+    super.setInstancePoker(Poker.class);
+    this.maxVal = 9;
+  }
+
+  public Ttl74192(String name, int maxVal) {
+    super(name, (byte) 16, OUTPUT_PORTS, PORT_NAMES, null);
+    super.setInstancePoker(Poker.class);
+    this.maxVal = maxVal;
+  }
+
+  public static class Poker extends InstancePoker {
+    boolean isPressed = true;
+
+    private boolean isInside(InstanceState state, MouseEvent e) {
+      final var p = getTranslatedTtlXY(state, e);
+      var inside = false;
+      for (var i = 0; i < 4; i++) {
+        final var dx = p.x - (56 + i * 10);
+        final var dy = p.y - 30;
+        final var d2 = dx * dx + dy * dy;
+        inside |= (d2 < 4 * 4);
+      }
+      return inside;
+    }
+
+    private int getIndex(InstanceState state, MouseEvent e) {
+      final var p = getTranslatedTtlXY(state, e);
+      for (var i = 0; i < 4; i++) {
+        int dx = p.x - (56 + i * 10);
+        int dy = p.y - 30;
+        int d2 = dx * dx + dy * dy;
+        if (d2 < 4 * 4) return 3 - i;
+      }
+      return 0;
+    }
+
+    @Override
+    public void mousePressed(InstanceState state, MouseEvent e) {
+      isPressed = isInside(state, e);
+    }
+
+    @Override
+    public void mouseReleased(InstanceState state, MouseEvent e) {
+      if (!state.getAttributeValue(TtlLibrary.DRAW_INTERNAL_STRUCTURE).booleanValue()) return;
+      if (isPressed && isInside(state, e)) {
+        var index = getIndex(state, e);
+        final var data = (TtlRegisterData) state.getData();
+        if (data == null) return;
+        var current = data.getValue().toLongValue();
+        final long bitValue = 1 << index;
```

Comment:
I guess you used `long` here because `var` was resolved to `int`? If that is so, then changing `1` to `1L` should make compiler correctly set it as `long`.

Reply:
This code block was copied from another TTL part.
I will create another PR where I fix this, [that](https://github.com/logisim-evolution/logisim-evolution/pull/1391#discussion_r813025326) and [that](https://github.com/logisim-evolution/logisim-evolution/pull/1391#discussion_r812603085) in multiple files.

Reply:
Perfect. I created #1392 to track this effort. Thanks

Reply:
@R3dst0ne feel free to mark this conversation as resolved if that'd help. The ticket is still there :)